pkgs/development/haskell-modules/lib/compose.nix

```diff
@@ -485,7 +485,7 @@ rec {
           builtins.genericClosure {
             startSet = builtins.map (drv:
               { key = drv.outPath; val = drv; }
-            ) drvs;
+            ) (builtins.filter (i: !builtins.isNull i) drvs);
```

Comment:
Is this necessary?

Reply:
yes, when it receives null it chokes so we need to filter it out, as mentioned in my other comment some derivations like gi-gmodule have a dependency set to null which means this function will error without this change.

Reply:
bb62d57a851 should also fix that – in general it is unlikely that `null` ends up in `libraryPkgConfigDepends` and it's correct, so keeping it as it was probably will catch similar problems in the future.